src/main/java/org/terasology/behaviors/system/FindNearbyPlayersSystem.java

```diff
@@ -66,12 +69,22 @@ public void update(float delta) {
                     continue;
                 }
                 if (locationComponent.getWorldPosition().distanceSquared(actorPosition) <= maxDistanceSquared) {
+                    if (charactersWithinRange.size() == 0) {
+                        closestCharacter = character;
+                        minDistanceFromCharacter = locationComponent.getWorldPosition().distanceSquared(actorPosition);
+                    } else {
+                        if (locationComponent.getWorldPosition().distanceSquared(actorPosition) < minDistanceFromCharacter) {
```

Comment:
If `minDistanceFromCharacter` gets initialized as 0.0, does this condition ever hold?

Reply:
Till it reaches 'else' branch, minDistanceFromCharacter would be updated as the distance of the first character to the actor. So, it wont be 0.0.